src/components/Editable.tsx

```diff
@@ -88,6 +88,7 @@ interface EditableProps {
   */
   transient?: boolean
   onEdit?: (args: { path: Path; oldValue: string; newValue: string }) => void
+  onThoughtFocus?: () => void
```

Comment:
If you are looking for a way to avoid all the props drilling, perhaps you can create a new ministore that the FauxCaret component can subscribe to, that can be directly updated from `Editable`. Or maybe you have another idea.